doc/source/users_guide/multi-instance.rst

```diff
@@ -1,95 +1,109 @@
+   user_nl_cism_0001 user_nl_mosart_0001
+   user_nl_cam_0002 user_nl_clm_0002 user_nl_docn_0002 user_nl_cice_0002
+   user_nl_cism_0002 user_nl_mosart_0002
    user_nl_cpl
 
-Also, **case.setup** creates the following ``*_in_*`` files and ``*txt*`` files in **$CASEROOT/CaseDocs**:
-::
-
-   atm_in_0001, atm_in_0002
-   docn.streams.txt.prescribed_0001, docn.streams.txt.prescribed_0002
-   docn_in_0001, docn_in_0002
-   docn_ocn_in_0001, docn_ocn_in_0002
-   drv_flds_in, drv_in
-   ice_in_0001, ice_in_0002
-   lnd_in_0001, lnd_in_0002
-   rof_in_0001, rof_in_0002
-
-The namelist for each component instance can be modified by changing the corresponding **user_nl_xxx_NNNN** file. 
-Modifying **user_nl_cam_0002** will result in your namelist changes being active ONLY for the second instance of CAM. 
+The namelist for each component instance can be modified by changing the corresponding **user_nl_xxx_NNNN** file.
+Modifying **user_nl_cam_0002** will result in your namelist changes being active ONLY for the second instance of CAM.
 To change the DOCN stream txt file instance 0002, copy **docn.streams.txt.prescribed_0002** to your **$CASEROOT** directory with the name **user_docn.streams.txt.prescribed_0002** and modify it accordlingly.
 
 Also keep these important points in mind:
 
+#. Note that these changes can be made at create_newcase time with option --ninst # where # is a positive integer, use the additional logical option --multi-driver to invoke the multi-driver mode.
+
 #. **Multiple component instances can differ ONLY in namelist settings; they ALL use the same model executable.**
 
-#. Multiple-instance implementation supports only one coupler component.
+#. Calling **case.setup** with ``--clean`` *DOES NOT* remove the **user_nl_xxx_NN** (where xxx is the component name) files created by **case.setup**.
+
+#. A special variable NINST_LAYOUT is provided for some experimental compsets, its value should be
+   'concurrent' for all but a few special cases and it cannot be used if MULTI_DRIVER=TRUE.
+
+#. In **create_test** these options can be invoked with testname modifiers _N# for the single driver mode and _C# for the multi-driver mode.  These are mutually exclusive options, they cannot be combined.
 
-#. Calling **case.setup** with ``--clean`` *DOES NOT* remove the **user_nl_xxx_NN** files created by **case.setup**.
+#. In create_newcase you may use --ninst # to set the number of instances and --multi-driver for multi-driver mode.
 
-#. Multiple instances generally should un concurrently, which is the default setting in **env_mach_pes.xml**. 
-   The serial setting is only for EXPERT USERS in upcoming development code implementations.
+#. In multi-driver mode you will always get 1 instance of each component for each driver/coupler, if you change a case using xmlchange MULTI_COUPLER=TRUE you will get a number of driver/couplers equal to the maximum NINST value over all components.
```

Comment:
@gold2718 this note was added to the documentation to clarify use of multi-driver mode.   In my opinion there is no error in what you did to create a case, there is however an expectation that the user understand how multi-driver differs from multi-instance and thus would expect a different list of user_nl_XXXX files.

Reply:
This seems confusion prone as `xmlquery NINST` does not show how many instances will actually run. @mvertens, @billsacks, any objections to this situation?

Reply:
@gold2718 Sorry, I haven't been following this whole thread. I'd be happy to weigh in if you'd be able to give a brief summary of the situation that you want input on.

Reply:
Look back about 6 posts to yesterday where I show two sets of commands (beginning with `./create_newcase`) and show that when `MULTI_DRIVER=TRUE`, the number of instances shown by `./xmlquery NINST` is not necessarily the number of instances which will be run.

Reply:
Sorry, I took some time to try to understand the current behavior and expected behavior, but I'm still a bit lost. Could we have a brief in-person chat about this on Tuesday?

Reply:
I don't want to take up a lot of your time with this but I will try a brief summary.
With a single driver, you can have N instances of some components but only one instance of others. For instance, you can have one ocean averaging the results of 64 atmospheres.

With multi-driver, each driver drives a separate instance of every non-stub component so if there are N drivers, there are N instances of every non-stub component.

The issue is that `xmlchange` allows you to set instance counts which are not viable with multi-driver but rather than display an error, the multi-driver code just ignores the situation and creates max(NINST_xxx) instances of every component. This behavior is mentioned at the end of the multi-instance documentation yet `xmlquery` still shows the values set by the user which is _not_ what is actually being run.

My question is whether this situation is okay or should be fixed. My concern is that users may read or hear that multi-driver speeds up the code so they turn it on. They can even check that they are running one ocean against 64 atmospheres (with `xmlquery`) but will in fact run 64 ocean instances with no averaging. As far as I can tell, there is no way to tell this is going to happen without running the case (or very carefully RTFM and we know how likely that is).

Reply:
If we decide that there should be a fix, I propose the following actions:

1. When setting `MULTI_DRIVER` from `FALSE` to `TRUE`, check to make sure all `NINST_xxx` have the same value and generate an error if not.
2. If `MULTI_DRIVER == TRUE`, then `case.setup` will check that all `NINST_xxx` have the same value and generate an error if not.

Reply:
@gold2718 thanks for your explanation. Now I understand the issue. Yes, I agree with you that this basic sanity checking should be done, rather than silently running something different from what was requested and/or listed values would suggest.